main.cpp

```diff
@@ -92,7 +92,128 @@ good to go!
  */
 
 #include <iostream>
-int main() 
+
+class FloatType
+{
+public:
+    FloatType()
+    {
+    }
+    ~FloatType()
+    {
+    }
+
+    float add( float lhs, float rhs )
+    {
+           return lhs + rhs;
+    }
+
+    float subtract( float lhs, float rhs )
+    {
+        return lhs - rhs;
+    }
+
+    float multiply( float lhs,float rhs )
+    {
+        return lhs * rhs;
+    }
+
+    float divide( float lhs,float rhs )
+    {
+        if (rhs == 0.0f)
+        std::cout << "\n" << "warning, floating point division by zero returns 'inf' !" << std::endl;
+        return lhs / rhs;
+    }
+
+private:
+
+};
+
+class DoubleType
+{
+public:
+    DoubleType()
+    {
+    }
+    ~DoubleType()
+    {
+    }
```

Comment:
Please use `struct`.  it's public by default
Save yourself some typing
- [ ] revise